### DISCUSSION TITLE
Grain growth from existing mesh file

### DISCUSSION THREAD

**Discussion by aashiquear** [idaholab/moose#17695] [Q&A Modules: Phase field] [answered]
I like to initialize a grain growth model from an existing mesh file (exodus format), that is already run a grain growth simulation for a while. I found there is an option in the PolycrystalVariables action "initial_from_file = true". Using that option I have setup a FileMeshGenerator Mesh and input the mesh file that I have already run for a while. This runs quite well for the first few steps but failed with an exit code 6 and a comment "Refinement flags were not consistent between processors!". Upon looking into the output I found, it initializes the grains from the previous file but once it started running there is instability around the four borders. Is there any example out there where it uses the "initial_from_file = true" in PolycrystalVariable action? Here is the file I am trying to run:

[grain_growth.txt](https://github.com/idaholab/moose/files/6388189/grain_growth.txt)

**Comment by permcody** [comment, marked as answer]
There are known issues with reading in a solution from an exodus file where adaptivity has been used. Generally we prefer to restart from either a checkpoint file or one of the libMesh formats (XDA/XDR). You might try outputting those and see if you can get it going (start with a small test case to avoid wasting time). Is this a restart case or are you creating a whole new simulation and just hoping to import some data from your previous simulation?

**Reply by aashiquear** [reply]
Right now I am trying a restart case, but eventually what I required is to run a whole new grain growth simulation using the grain structure generated from the previous file. I am running a simultaneous grain growth and another physics using a multiapp to see the effect of the other physics when grain migrated. If I start from voronoi IC the grain growth starts slow as it equilibrates to the GB width first and I failed to capture the initial effect as my other physics thinking the grain structure as static in the beginning. So, what I am trying to capture here is that both the script in multiapp calibrated in such that the initial GB width equilibration doesn't require and grain migration starts right away. After yesterday, I have run the same file I submit here using a single processor and can run the full simulation without failing. I get what I was looking for but there is instability (like the order parameter fluctuates) around the domain border still present. I am going to check with outputting in XDA format and re-reading the file in the new simulation. Please share your idea if there is a better (and easy) way of doing the same. Thanks

**Reply by permcody** [reply]
There is some low hanging fruit in terms of optimizing and fixing some of the initial condition behavior of simulations with large numbers of variables. We already have this on our todo list, however being aware that others are needing this may help to bump priority. 

@fdkong
